R/zchunk_LA131.enduse.R

```diff
@@ -35,58 +36,189 @@ module_energy_LA131.enduse_DISABLED <- function(command, ...) {
+    A_regions %>%
+      filter(heat == "1") %>%
+      select(GCAM_region_ID) %>%
+      .$GCAM_region_ID ->
+      GCAM_region_ID_heat
+
+    Enduse_elect_scaled_heat_unscaled %>%
+      filter(fuel == "heat", GCAM_region_ID %in% GCAM_region_ID_heat) %>%
+      group_by(GCAM_region_ID, year) %>%
+      summarise(value = sum(value)) ->
+      Enduse_heat_unscaled
+
+    # Calculate the scalers required to balance district heat production and consumption within each region
+    Enduse_heat %>%
+      left_join_error_no_match(Enduse_heat_unscaled, by = c("GCAM_region_ID", "year")) %>%
+      mutate(value = value.x / value.y) %>%
+      mutate(value = if_else(is.na(value),0,value)) %>%
+      select(-value.x, -value.y) ->
+      Enduse_heat_scaler
+
+    # Multiply the district heat scalers by the original estimates of district heat consumption by end use sectors
+    Enduse_elect_scaled_heat_unscaled %>%
+      filter(fuel == "heat", GCAM_region_ID %in% GCAM_region_ID_heat) %>%
+      left_join_error_no_match(Enduse_heat_scaler, by = c("GCAM_region_ID", "year")) %>%
+      mutate(value = value.x * value.y) %>%
+      select(-value.x, -value.y) ->
+      Enduse_heat_scaled
+
+    # Replace unscaled estimates of end use sector heat consumption in full table
+    Enduse_elect_scaled_heat_unscaled %>%
+      mutate(fuel = replace(fuel, GCAM_region_ID %in% GCAM_region_ID_heat & fuel == "heat", "heat2")) %>%
+      filter(fuel != "heat2") %>%
+      bind_rows(Enduse_heat_scaled) %>%
+      arrange(GCAM_region_ID, sector, fuel, year) ->
+      L131.in_EJ_R_Senduse_F_Yh # Output table 1
+
+    # Heat in some regions is not modeled separately from the fuels used to produce it
+    A_regions %>%
+      filter(heat == "0") %>%
+      select(GCAM_region_ID) %>%
+      .$GCAM_region_ID ->
+      GCAM_region_ID_no_heat
+
+    # In these regions, calculate the share of regional heat demand by each sector
+    L131.in_EJ_R_Senduse_F_Yh %>%
+      group_by(GCAM_region_ID, fuel, year) %>%
+      summarise(value = sum(value)) ->
+      Enduse_total
+
+    L131.in_EJ_R_Senduse_F_Yh %>%
+      filter(fuel == "heat") %>%
+      filter(GCAM_region_ID %in% GCAM_region_ID_no_heat) %>%
+      left_join_error_no_match(Enduse_total, by = c("GCAM_region_ID", "fuel", "year")) %>%
+      mutate(value = value.x / value.y) %>%
+      select(-value.x, -value.y) ->
+      Enduse_heat_scaled_share
+
+    # Regions may have zero heat consumption by demand sectors while nevertheless having heat production. Assign this to industry
+    Enduse_heat_scaled_share %>%
+      filter(sector == "in_industry_general") %>%
+      mutate(value = if_else(is.na(value),1,value)) ->
+      Enduse_heat_scaled_share_indust
+
+    Enduse_heat_scaled_share %>%
+      filter(sector != "in_industry_general") %>%
+      mutate(value = if_else(is.na(value),0,value)) %>%
+      bind_rows(Enduse_heat_scaled_share_indust) %>%
+      arrange(GCAM_region_ID, sector, year) ->
```

Comment:
In addition to being ordered on the level1 data output for visual purposes, sometimes ordering has real consequences for how GCAM overrides read in data. We have some input xml files that unintentionally overrides later year data with previous year data because the years are out of order. I think it's a good thing to use arrange.